master/internal/config/config.go

```diff
@@ -328,8 +328,18 @@ func (c *Config) Resolve() error {
 	}
 
 	for _, r := range c.ResourceManagers() {
-		if r.ResourceManager.AgentRM != nil && r.ResourceManager.AgentRM.Scheduler == nil {
-			r.ResourceManager.AgentRM.Scheduler = DefaultSchedulerConfig()
+		if r.ResourceManager.AgentRM != nil {
+			if r.ResourceManager.AgentRM.Scheduler == nil {
+				r.ResourceManager.AgentRM.Scheduler = DefaultSchedulerConfig()
+			}
+			if r.ResourceManager.AgentRM.Scheduler.GetType() == FairShareScheduling {
+				log.Warn("Fair-Share Scheduler has been deprecated, please update master config to use Priority Scheduler.")
+			}
+			if r.ResourceManager.AgentRM.Scheduler.GetType() == RoundRobinScheduling {
+				log.Error("Round Robin Scheduler has been removed, please update master config to use Priority Scheduler.")
+				log.Info("Priority Scheduler with all priorities equal will have the same behavior as a Round Robin Scheduler.")
+				return fmt.Errorf("scheduler not available")
+			}
```

Comment:
I think the changes in ```MakeScheduler``` should take care of this right?

I think this would log two warnings if you specified fairshare in the resource manager level

Reply:
The changes in `MakeScheduler` do take care of it, for the most part.

If the resource manger is configured to use `round_robin` but all resource pools are configured to use other schedulers (like below), then `MakeScheduler` will not complain. Do we care about that (admittedly unlikely) case?
```
        resource_manager:
          type: agent
          scheduler:
            type: round_robin
            fitting_policy: best
        resource_pools:
        - pool_name: pool1
          scheduler:
            type: fair_share
            fitting_policy: best
```

Reply:
I think it is probably fine to not handle that case.

Though if we do want to handle that case it would be nice to do it without logging more than expected warnings.